homeassistant/components/fritz/common.py

```diff
@@ -0,0 +1,270 @@
+                                    vol.Required(CONF_PASSWORD): cv.string,
+                                }
+                            )
+                        ],
+                    )
+                }
+            )
+        },
+    ),
+    extra=vol.ALLOW_EXTRA,
+)
+
+SERVICE_SCHEMA = vol.Schema({vol.Required(ATTR_HOST): cv.string})
+
+
+class FritzBoxTools:
+    """FrtizBoxTools class."""
+
+    def __init__(
+        self,
+        hass,
+        password,
+        username=DEFAULT_USERNAME,
+        host=DEFAULT_HOST,
+        port=DEFAULT_PORT,
+    ):
+        """Initialize FritzboxTools class."""
+        self._device_info = None
+        self._devices: Dict[str, Any] = {}
+        self._unique_id = None
+        self.connection = None
+        self.error = None
+        self.fritzhosts = None
+        self.fritzstatus = None
+        self.ha_ip = None
+        self.hass = hass
+        self.host = host
+        self.password = password
+        self.port = port
+        self.success = None
+        self.username = username
+
+    async def async_setup(self):
+        """Wrap up FritzboxTools class setup."""
+        return await self.hass.async_add_executor_job(self.setup)
+
+    def setup(self):
+        """Set up FritzboxTools class."""
+        try:
+            self.connection = FritzConnection(
+                address=self.host,
+                port=self.port,
+                user=self.username,
+                password=self.password,
+                timeout=60.0,
+            )
+
+            self.fritzstatus = FritzStatus(fc=self.connection)
+            self._unique_id = self.connection.call_action("DeviceInfo:1", "GetInfo")[
+                "NewSerialNumber"
+            ]
+            self.fritzhosts = FritzHosts(fc=self.connection)
+            self._device_info = self._fetch_device_info()
+            self.success = True
+            self.error = False
+        except FritzConnectionException:
+            self.success = False
+            self.error = ERROR_CONNECTION_ERROR
+
+        self.ha_ip = get_local_ip()
+
+        self.scan_devices()
+
+        async_track_time_interval(
+            self.hass, self.scan_devices, timedelta(seconds=TRACKER_SCAN_INTERVAL)
+        )
+
+        return self.success, self.error
+
+    @property
+    def unique_id(self):
+        """Return unique id."""
+        return self._unique_id
+
+    @property
+    def fritzbox_model(self):
+        """Return model."""
+        return self._device_info["model"].replace("FRITZ!Box ", "")
+
+    @property
+    def device_info(self):
+        """Return device info."""
+        return self._device_info
+
+    @property
+    def devices(self) -> Dict[str, Any]:
+        """Return devices."""
+        return self._devices
+
+    @property
+    def signal_device_new(self) -> str:
+        """Event specific per Fritzbox entry to signal new device."""
+        return f"{DOMAIN}-device-new"
```

Comment:
Dispatcher signals are global, if they have more than one of these you'll get dispatches between config entries.  Include the unique id for the entry in the signal.

Reply:
Fixed